python-frontend/src/main/java/org/sonar/python/semantic/SymbolTableBuilder.java

```diff
@@ -132,13 +133,69 @@ public void visitFileInput(FileInput fileInput) {
     scopesByRootTree = new HashMap<>();
     fileInput.accept(new FirstPhaseVisitor());
     fileInput.accept(new SecondPhaseVisitor());
+    createAmbiguousSymbols();
     addSymbolsToTree((FileInputImpl) fileInput);
     fileInput.accept(new ThirdPhaseVisitor());
     if (!SymbolUtils.isTypeShedFile(pythonFile)) {
       TypeInference.inferTypes(fileInput);
     }
   }
 
+  private static class SymbolToUpdate {
+    final Symbol symbol;
+    final AmbiguousSymbol ambiguousSymbol;
+
+    SymbolToUpdate(Symbol symbol, AmbiguousSymbol ambiguousSymbol) {
+      this.symbol = symbol;
+      this.ambiguousSymbol = ambiguousSymbol;
+    }
+  }
+
+  private void createAmbiguousSymbols() {
+    for (Scope scope : scopesByRootTree.values()) {
+      Set<SymbolToUpdate> symbolsToUpdate = new HashSet<>();
+      for (Symbol symbol : scope.symbols()) {
+        if (symbol.kind() == Symbol.Kind.OTHER) {
+          List<Usage> bindingUsages = symbol.usages().stream().filter(Usage::isBindingUsage).collect(Collectors.toList());
+          if (bindingUsages.size() > 1) {
```

Comment:
I think that this condition is not sufficient. We end up creating an ambiguous symbol in the following case and I suppose we should not:
```
x = 42
x = 43
```

Reply:
Good point. I'll update the condition to also check the presence of a usage of kind `FUNC_DECLARATION` or `CLASS_DECLARATION`